nunaliit2-js/src/main/js/nunaliit2/n2.widgetTime.js

```diff
@@ -376,44 +873,74 @@ var TimelineWidget = $n2.Class({
 	}
 });
 
-//--------------------------------------------------------------------------
+// --------------------------------------------------------------------------
 function HandleWidgetAvailableRequests(m){
-	if( m.widgetType === 'timeline' ){
+	if( m.widgetType === 'dateRange' ){
 		if( $.fn.slider ) {
 			m.isAvailable = true;
-		};
-    };
+		}
+    } else if( m.widgetType === 'timeline' ){
+		if( $.fn.slider ) {
+			m.isAvailable = true;
+		}
+    }
 };
 
-//--------------------------------------------------------------------------
+// --------------------------------------------------------------------------
 function HandleWidgetDisplayRequests(m){
-	if( m.widgetType === 'timeline' ){
-		var widgetOptions = m.widgetOptions;
-		var containerId = m.containerId;
-		var config = m.config;
-		
-		var options = {};
+	var widgetOptions, config, options, optionKeys, containerId, i, key, value;
+	if (m.widgetType === 'dateRange') {
+		widgetOptions = m.widgetOptions;
+		containerId = m.containerId;
+		config = m.config;
+		options = {};
+
+		if (widgetOptions) {
```

Comment:
Much of this section is common too.  For simplicity and maintenance, I suggest moving common code outside the if/else statements.

Reply:
Fixed!